src/main/java/lesson2/problem6/MergeKLists.java

```diff
@@ -0,0 +1,69 @@
+package lesson2.problem6;
+
+import lesson2.problem0.Link;
+import lesson2.problem0.LinkedList;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.PriorityQueue;
+
+public class MergeKLists {
+
+    public static Link mergeLists(List<Link> input) {
+        if (input == null || input.size() == 0) {
+            throw new IllegalArgumentException();
+        }
+        final PriorityQueue<Link> myQueue = new PriorityQueue<>(input.size(), new Comparator<Link>() {
+            @Override
+            public int compare(Link o1, Link o2) {
+                return 0;
+            }
+        });
+
+        Link returnList = new Link(0);
+        Link listEnd = returnList;
+
+        for (Link thisLink : input) {
+            if (thisLink != null) {
+                myQueue.add(thisLink);
```

Comment:
No, we do not add all the items to the queue. The queue should hold K elements at the one time. You should get the minimum from each linked list, which is the head of each initially. Then remove the minimum from the queue, and move the next element in that linked list that provided the minimum and put that element in the queue and repeat.

Reply:
given an input `List<Link> input` this for loop will cycle through and add only one node (AKA the head) per Node in the list. Therefore, I believe K nodes are added to the queue through this loop